### PR TITLE
feat(multichain): add multichain support

### PR DESCRIPTION
**Pull request by nutrina**
This addresses: #739 

- allow users to connect to multiple chains
- keep using a mainnet did for the passport session, regardless of chain

### CODE REVIEW

File: app/context/userContext.tsx [hunk 1]
```diff
@@ -125,22 +129,54 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
           const sessionKey = `didsession-${address}`;
-          const sessionStr = localStorage.getItem(sessionKey);
+          const sessionStr = localStorage.getItem(sessionKey + "BAD");
 
           // @ts-ignore
-          let selfId = await ceramicConnect(ethAuthProvider, sessionStr);
+          // When sessionStr is null, this will create a new selfId. We want to avoid this, becasue we want to make sure
+          // that chainId 1 is in the did
+          let selfId = !!sessionStr ? await ceramicConnect(ethAuthProvider, sessionStr) : null;
+          console.log(
```
Comment (aminah-io) [inline app/context/userContext.tsx:138]:
```suggestion

```

File: app/context/userContext.tsx [hunk 2]
```diff
@@ -125,22 +129,54 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
+            "geri - process.env.NEXT_PUBLIC_FF_MULTICHAIN_SIGNATURE",
```
Comment (aminah-io) [inline app/context/userContext.tsx:139]:
```suggestion

```

File: app/context/userContext.tsx [hunk 3]
```diff
@@ -125,22 +129,54 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
+            process.env.NEXT_PUBLIC_FF_MULTICHAIN_SIGNATURE
```
Comment (aminah-io) [inline app/context/userContext.tsx:140]:
```suggestion

```

File: app/context/userContext.tsx [hunk 4]
```diff
@@ -125,22 +129,54 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
+          );
```
Comment (aminah-io) [inline app/context/userContext.tsx:141]:
```suggestion

```

File: app/context/userContext.tsx [hunk 5]
```diff
@@ -125,22 +129,54 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
+              // Also we enforce the "1" chainId, as we always want to use mainnet dids, in order to avoid confusion
+              // as to where a passport / stamp has been stored
+              const authMethod = await EthereumWebAuth.getAuthMethod(
+                wallet.provider,
+                new AccountId({
+                  chainId: "eip155:1",
+                  address: address,
+                })
+              );
+              console.log("geri - authMethod", authMethod);
```
Comment (aminah-io) [inline app/context/userContext.tsx:165]:
```suggestion

```

File: app/context/userContext.tsx [hunk 6]
```diff
@@ -125,22 +129,54 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
+              const session = await DIDSession.authorize(authMethod, {
+                expiresInSecs: 24 * 3600,
+                resources: ["ceramic://*"],
+              });
+              const newSessionStr = session.serialize();
+
+              selfId = await ceramicConnect(ethAuthProvider, newSessionStr);
+            } else {
+              console.log("geri - NEW default selfId");
```
Comment (aminah-io) [inline app/context/userContext.tsx:174]:
```suggestion

```

File: app/context/userContext.tsx [hunk 7]
```diff
@@ -125,22 +129,54 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
+              // If the session loaded is not valid, or if it is expired or close to expire, we create
+              // a new connection to ceramic
+              selfId = await ceramicConnect(ethAuthProvider);
+              console.log("geri - selfId", selfId);
```
Comment (aminah-io) [inline app/context/userContext.tsx:178]:
```suggestion

```

**Review by aminah-io — APPROVED** [review]
Just some log removals, otherwise LGTM!

File: app/context/userContext.tsx [hunk 8]
```diff
@@ -125,22 +129,47 @@ export const UserContextProvider = ({ children }: { children: any }) => {
...
             !selfId?.client?.session ||
             // @ts-ignore
             selfId?.client?.session?.isExpired ||
             // @ts-ignore
             selfId?.client?.session?.expireInSecs < 3600
           ) {
-            // If the session loaded is not valid, or if it is expired or close to expire, we create
-            // a new connection to ceramic
-            selfId = await ceramicConnect(ethAuthProvider);
+            if (process.env.NEXT_PUBLIC_FF_MULTICHAIN_SIGNATURE === "on") {
```
Comment (schultztimothy) [inline app/context/userContext.tsx:149]: Maybe write some tests around the edge cases here?

File: app/context/userContext.tsx [hunk 9]
```diff
@@ -125,22 +129,47 @@ export const UserContextProvider = ({ children }: { children: any }) => {
           // Sessions will be serialized and stored in localhost
           // The sessions are bound to an ETH address, this is why we use the address in the session key
           const sessionKey = `didsession-${address}`;
-          const sessionStr = localStorage.getItem(sessionKey);
+          const sessionStr = localStorage.getItem(sessionKey + "BAD");
```
Comment (schultztimothy) [inline app/context/userContext.tsx:132]: This prolly needs removed?

Comment (schultztimothy) [reply app/context/userContext.tsx:132]: Works great when pulling everything down locally!

File: app/context/userContext.tsx (hunk 1 above)
Comment (nutrina) [reply app/context/userContext.tsx]: Done

**Comment by nutrina** [comment]
This is now deprecated by #822

**Comment by chibie** [comment]
can we close this PR? @nutrina

**Comment by nutrina** [comment]
Closing the PR, as this was already merged from another branch.
